tests/test_tube_particles.py

```diff
@@ -0,0 +1,85 @@
+from __future__ import print_function
+
+import numpy as np
+import matplotlib.pyplot as plt
+import dxchange
+
+from xdesign.material import XraylibMaterial, CustomMaterial
+from xdesign.geometry import *
+from xdesign.phantom import Phantom
+from xdesign.propagation import *
+from xdesign.plot import *
+from xdesign.acquisition import Simulator
+
+
+def test_model_prop_pipeline():
+
+    n_particles = 20
+    top_y = 50.
+    top_radius = 20.
+    bottom_radius = 200.
+    top_thickness = 10.
+    bottom_thickness = 30.
+    length = 400.
+    bottom_y = top_y + length
+
+    silicon = XraylibMaterial('Si', 2.33)
+    titania = XraylibMaterial('TiO2', 4.23)
+    air = CustomMaterial(delta=0, beta=0)
+
+    try:
+        grid_delta = np.load('data/sav/grid/grid_delta.npy')
+        grid_beta = np.load('data/sav/grid/grid_beta.npy')
+    except:
```

Comment:
The structure is just a alternative to an `if` judgement: skip modeling if data files exist, otherwise create model and generate grids.